lib/adagrams.rb

```diff
@@ -0,0 +1,89 @@
+require "csv"
+
+# Helper method that creates array of letters that are available to be drawn
+def array_gen(hash)
+  array = []
+  hash.each do |key, value|
+    value.times do
+      array << key.to_s
+    end
+  end
+  return array
+end
+
+# Returns an array of 10 letters chosen at random from available letters
+def draw_letters
+  letter_freq = {
+    A: 9, N: 6, B: 2, O: 8, C: 2, P: 2, D: 4, Q: 1, E: 12, R: 6, F: 2, S: 4,
+    G: 3, T: 6, H: 2, U: 4, I: 9, V: 2, J: 1, W: 2, K: 1, X: 1, L: 4, Y: 2, M: 2, Z: 1,
+  }
+
+  avail_letters = array_gen(letter_freq)
+  used_letters = avail_letters.sample(10)
```

Comment:
This is lovely! I can see everything I need to see about how this works, and if I get confused about `array_gen` it's close at hand!